onnxruntime/core/optimizer/constant_folding.cc

```diff
@@ -90,6 +90,71 @@ static bool ConstantFoldShapeNode(Graph& graph, Node& node) {
   return is_concrete_shape;  // convert to constant if this is true
 }
 
+// This function inlines the appropriate subgraph. It does not literally fold it.
+static Status ConstantFoldIfNode(Graph& graph, Node& if_node, const logging::Logger& logger, bool& folded) {
+  folded = false;
+  // First, find out which subgraph to inline
+  // We need to fetch the constant argument.
+  ORT_ENFORCE(if_node.InputDefs().size() == 1, "Expecting one input");
+  const auto* condition_def = if_node.InputDefs()[0];
+
+  // We need to check if the condition is a constant.
+  constexpr bool check_outer_scope_true = true;
+  const ONNX_NAMESPACE::TensorProto* initializer = graph.GetConstantInitializer(condition_def->Name(), check_outer_scope_true);
+  if (initializer == nullptr) {
+    return Status::OK();
+  }
+
+  // This is a boolean initializer with a single element.
+  const auto proto_shape = utils::GetTensorShapeFromTensorProto(*initializer);
+  // This must be pre-allocated
+  Tensor tensor{
+      DataTypeImpl::TensorTypeFromONNXEnum(initializer->data_type())->GetElementType(),
+      proto_shape,
+      std::make_shared<CPUAllocator>()};
+
+  Status status = utils::TensorProtoToTensor(Env::Default(), graph.ModelPath().ToPathString().c_str(), *initializer, tensor);
```

Comment:
I will see what I can do. I lint everything before pushing changes, otherwise CI fails.

Reply:
Linting doesn't (currently) enforce the 120 char limit. You need to manually do that - which is why having a vertical ruler in your editor makes it way easier to do when editing the code. clang-format does a _horrible_ job if you let it it will throw away all rational formatting and try and cram as much as possible on every line which completely screws up readability.

I do have a script to apply clang-format only to long lines that could maybe be plugged in to provide automatic suggestions.

Reply:
Just for you: https://github.com/microsoft/onnxruntime/pull/18293